### DISCUSSION TITLE
Using cbl in multiple flutter screens

### DISCUSSION THREAD

**Discussion by gruuna** [cofu-app/cbl-dart#542] [General]
We're having the requirement that our app has to support multiple instances to use different screens of our app in split-view on iOS. Following https://docs.flutter.dev/add-to-app/multiple-flutters there is a solution with multiple flutter engines connected by a flutter-engine-group.

I've already recognized that there is a CouchbaseLite.initSecondary method for using cbl in multiple isolates - but in case of multiple flutter engines I think that the needed context cannot be shared, am I right?

Is there an other solution and would it be also possible to do replications in this scenario?

**Comment by blaugold** [comment]
Hm, good question. I think there currently might be one or two issues blocking this use case. I'll set up an app using multiple Flutter Engines to evaluate what kind of work might be necessary.

My feeling is that, it should be possible to make this use case work.

**Comment by gruuna** [comment]
I've tested a bit with our app where I introduced scene functionality and I figured out that CBL is (almost) working when I set the custom logger to null:

`Database.log
        ..custom = null 
        );`

With that set, I don't get any errors that a custom logger is already set in isolate.... but I don't have any loggings for sure :-)

Are there any other methods to set a logger and do you think CBL is really working in that scenario? Because I'm not sure what's happening now... is there only 1 local couchbase and what about running replications?

**Reply by blaugold** [reply]
Sorry for the late response. 😔 How logging currently works is what I thought might be a problem. You should at least be able to use [File logging](https://pub.dev/documentation/cbl/latest/cbl/Log/file.html) until there is a fix.

There should be no issue using Couchbase Lite from two different screens. If you open the same database file from multiple screens, they will be able to read at the same time, but for writing they will have to wait for each other. But this synchronization is all handled for you by Couchbase Lite. You can run two replicators in parallel on the same database file, so that should also not be an issue.

I've created an issue to track supporting your overall use case (https://github.com/cbl-dart/cbl-dart/issues/543).
